Framework/API/src/WorkspaceOpOverloads.cpp

```diff
@@ -412,22 +412,26 @@ bool WorkspaceHelpers::commonBoundaries(const MatrixWorkspace &WS) {
     return true;
 
   // But even if they're not they could still match...
-  const double commonSum =
-      std::accumulate(WS.readX(0).begin(), WS.readX(0).end(), 0.);
+  const auto &x_0 = WS.readX(0);
+  const double commonSum = std::accumulate(x_0.begin(), x_0.end(), 0.);
   // If this results in infinity or NaN, then we can't tell - return false
   if (!std::isfinite(commonSum))
     return false;
   const size_t numHist = WS.getNumberHistograms();
   for (size_t j = 1; j < numHist; ++j) {
-    const double sum =
-        std::accumulate(WS.readX(j).begin(), WS.readX(j).end(), 0.);
+    const auto &x_j = WS.readX(j);
+    // they should all have the same number of x-values
+    if (x_0.size() != x_j.size())
+      return false;
+
+    const double sum = std::accumulate(x_j.begin(), x_j.end(), 0.);
```

Comment:
As probably mentioned in the past, what `commonBoundaries` does is conceptually broken. Would it be in the scope of this PR to fix this?

Reply:
As you mentioned on Slack: Out of scope. :+1:

Reply:
Maybe also here a length check should be added? Otherwise having variable-length histograms introduces new failure modes to the method, e.g., if you have a workspace with two histograms with bin edges `{1,2,4}` and `{3,4}`.